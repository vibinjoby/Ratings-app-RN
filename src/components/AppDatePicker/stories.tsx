import React from 'react'
import { storiesOf } from '@storybook/react-native'
import AppDatePicker from './component'

export const Story = () => <AppDatePicker />

storiesOf('Components', module).add('AppDatePicker', () => <Story />)

export default {
  title: 'AppDatePicker',
}
