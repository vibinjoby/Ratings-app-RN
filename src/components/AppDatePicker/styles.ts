import { StyleSheet } from 'react-native'
import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {},
  textInput: {
    fontFamily: 'Apercu-Light',
    fontSize: 14,
    padding: 10,
    borderTopLeftRadius: 4,
    borderBottomLeftRadius: 4,
    alignItems: 'flex-start',
    color: Colors.textInputBorder,
    borderColor: Colors.datePickerBorder,
    borderWidth: 1,
    borderRightWidth: 0,
  },
  dateIcContainer: {
    borderWidth: 1,
    borderTopRightRadius: 4,
    borderBottomRightRadius: 4,
    padding: 9,
    borderColor: Colors.datePickerBorder,
  },
  dateIcon: {
    width: 20,
    height: 20,
    marginLeft: -1,
    borderColor: Colors.datePickerBorder,
  },
  calendarImg: {
    width: 20,
    height: 20,
  },
  appTxt: {
    ...Typography.RegularFont,
    fontSize: 16,
    color: Colors.black,
    marginBottom: 10,
  },
  datePicker: {
    width: '100%',
    textAlign: 'left',
  },
})
