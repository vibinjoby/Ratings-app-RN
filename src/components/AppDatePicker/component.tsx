import React, { useState } from 'react'
import { View, Text, Image, ViewProps } from 'react-native'
import DatePicker from 'react-native-datepicker'
import moment from 'moment'

import styles from './styles'

export interface AppDatePickerProps {
  label?: string
  defaultDate?: string
  customStyle?: ViewProps['style']
  onDateSelection?: (arg0: string) => void
  selectedDay?: () => number
  containerStyle?: ViewProps['style']
  isPastAllowed?: boolean
  format?: string
}

const AppDatePicker: React.FC<AppDatePickerProps> = ({
  label,
  customStyle,
  onDateSelection,
  containerStyle,
  format = 'Do MMM,YYYY',
}: AppDatePickerProps) => {
  const [date, setDate] = useState(new Date())

  const validateDateChange = (_: Event, date: Date | undefined) => {
    if (!date) return

    setDate(date)
    onDateSelection && onDateSelection(date.toDateString())
  }

  const DateIconComp = () => (
    <View style={styles.dateIcContainer}>
      <Image source={require('../../assets/calendar.png')} style={styles.calendarImg} />
    </View>
  )
  return (
    <View style={containerStyle}>
      <Text style={styles.appTxt}>{label}</Text>

      <DatePicker
        testID="datePicker"
        style={[styles.datePicker, customStyle]}
        mode="date"
        placeholder="Select a date"
        date={date}
        maxDate={moment().toDate()}
        format={format}
        confirmBtnText="Confirm"
        cancelBtnText="Cancel"
        iconComponent={<DateIconComp />}
        customStyles={{
          dateInput: styles.textInput,
          datePicker: {
            backgroundColor: '#d1d3d8',
            justifyContent: 'center',
          },
        }}
        onDateChange={validateDateChange}
      />
    </View>
  )
}

export default AppDatePicker
