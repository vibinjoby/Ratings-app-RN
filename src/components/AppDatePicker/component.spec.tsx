import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import AppDatePicker from './component'

describe('AppDatePicker', () => {
  const MOCK_FN = jest.fn()

  it('should render', () => {
    const { toJSON } = render(<AppDatePicker />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should be able to click on the date picker and change date', () => {
    const today = new Date()
    const { getByTestId } = render(<AppDatePicker onDateSelection={MOCK_FN} />)
    fireEvent(getByTestId('datePicker'), 'onDateChange', today)
    expect(MOCK_FN).toBeCalledWith(today)
  })
})
