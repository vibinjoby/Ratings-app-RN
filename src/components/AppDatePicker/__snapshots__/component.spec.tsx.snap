// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDatePicker should render 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#181725",
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "width": 142,
        },
        Array [
          Object {
            "textAlign": "left",
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
    testID="datePicker"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#aaa",
              "borderWidth": 1,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "flex-start",
              "borderBottomLeftRadius": 4,
              "borderColor": "rgba(84, 101, 233, 0.2)",
              "borderRightWidth": 0,
              "borderTopLeftRadius": 4,
              "borderWidth": 1,
              "color": "rgba(30, 33, 76,0.1)",
              "fontFamily": "Apercu-Light",
              "fontSize": 14,
              "padding": 10,
            },
            false,
            false,
          ]
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#c9c9c9",
              },
              undefined,
            ]
          }
        >
          Select a date
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomRightRadius": 4,
            "borderColor": "rgba(84, 101, 233, 0.2)",
            "borderTopRightRadius": 4,
            "borderWidth": 1,
            "padding": 9,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/calendar.png",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  pointerEvents="auto"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderTopColor": "#ccc",
                          "borderTopWidth": 1,
                          "marginTop": 42,
                        },
                        Object {
                          "backgroundColor": "#d1d3d8",
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <RCTDatePicker
                      date={1630766928346}
                      maximumDate={1630766928346}
                      mode="date"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 216,
                        }
                      }
                      timeZoneOffsetInMinutes={null}
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#666",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
